alkemy/src/main/kotlin/io/alkemy/AlkemyExtension.kt

```diff
@@ -0,0 +1,45 @@
+package io.alkemy
+
+import io.alkemy.config.AlkemyConfig
+import io.kotest.core.TestConfiguration
+import io.kotest.core.extensions.MountableExtension
+import io.kotest.core.extensions.install
+import io.kotest.core.listeners.AfterSpecListener
+import io.kotest.core.spec.Spec
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+fun Spec.installAlkemyExtension(
```

Comment:
Apologies for the delay, I only managed to have a look at this today. 
I believe we can live without the constructor injection, and your other arguments around compatibility (and lack of friction) with other tools seem sound.

A couple of things we'd need to consider though. 

I think we can create better naming for `installAlkemyExtension`. I think it's not entirely intuitive and it also leaks what the method does from a call-site perspective (create or fetch a context). The "pooled" flag is also simply not intuitive enough and it might be worth having two methods `defaultAlkemyContext` and `customAlkemyContext`.

But before we even get to that, more importantly, the core behavioural difference here is the fact that the extension is installed on a per-test basis, as far as I can tell. This sub-optimal if the webdrivers get killed and opened again with each test. One of the benefits of the existing approach is that these are re-used and there's less of a performance penalty on bootstrapping the browser(s).

It's a core principle of Alkemy that you configure these drivers at bootstrap (for most cases, at least), and then re-use them, which can help run the tests really quickly. This is to me, something I don't think we can make a concession on. But do let me know if I misunderstood some behaviour here.

Reply:
> I think we can create better naming for installAlkemyExtension. I think it's not entirely intuitive and it also leaks what it actually does vs what the call-site needs (create or fetch a context)

Sure

> But before we even get to that, more importantly, the core behavioural difference here is the fact that the extension is installed on a per-test (spec) basis, as far as I can tell. This sub-optimal if the webdrivers get killed and opened again with each test. One of the benefits of the existing approach is that these are re-used and there's less of a performance penalty on bootstrapping the browser(s).

> It's a core principle of Alkemy that you configure these drivers at bootstrap (for most cases, at least), and then re-use them, which can help run the tests really quickly. This is to me, something I don't think we can make a concession on. But do let me know if I misunderstood some behaviour here.

You're right, that's a huge tradeoff. I couldn't figure out how to separate the pool for default configurations and custom configurations. Your suggestion of having 2 separate methods (`defaultAlkemyContext` and `customAlkemyContext`) solves the problem, I'll make the change, thanks!

Reply:
I found a better way. I forgot that I previously already enhanced the `WebDriverPool` to support mutiple drivers with different configurations (by using the `AlkemyConfig` as the hash key to lookup drivers), so all that was needed to share the `WebDriverPool` between Specs was to move the `WebDriverPool` to the Project level, which I've done by adding `AlkemyWebDriverPoolExtension` that is auto-loaded by Kotest using `@AutoScan`.

I've removed the `pooled` flag, since even custom configs are supported by the enhanced `WebDriverPool`. Now `AlkemyContext.PooledDriver` is the default for both default/custom configs, and `AlkemyContext.NewDriver` is only used if the  `AlkemyWebDriverPoolExtension` is not loaded (Kotest auto scan is disabled and the extension is not manually loaded).